fix the target image of openshift naming

jianzhangbjz opened the pull request:
I cherry pick the #83, thanks all!

dymurray commented on registries/adapters/openshift_adapter.go:
```diff
@@ -43,7 +42,7 @@ type OpenShiftImage struct {
 // RegistryName - Retrieve the registry name
 func (r OpenShiftAdapter) RegistryName() string {
-	return openShiftName
+	return strings.TrimPrefix(r.Config.URL.String(), "https://")
```
what if I use `http://<url>` as my URL? Would prefer to see it validate the string first

jmrodri replied:
This only removes https, if the url has any other prefix this won't work.

jmrodri commented on registries/adapters/openshift_adapter_test.go:
```diff
@@ -28,6 +28,8 @@ import (
 
 var Images = []string{"satoshi/nakamoto", "foo/bar", "paul/atreides"}
+var TestURL = "https://registry.reg-aws.openshift.com:443"
+var RegistryName = "registry.reg-aws.openshift.com:443"
```
No need to declare these strings here. I would localize them to the test case instead.

dymurray commented:
https://github.com/automationbroker/bundle-lib/pull/86

Submitted a PR to master to attempt to address `http://`

jmrodri commented on registries/adapters/openshift_adapter_test.go:
```diff
@@ -89,9 +91,17 @@ const AuthResponse = `
+	url, err := url.Parse(TestURL)
+	if err != nil {
+		t.Fatal("Error: ", err)
+	}
```
Consider using a table for the test cases, this will allow us to easily add new urls if we want to test other conditions:
```golang
func TestRegistryName(t *testing.T) {

    testcases := []struct {
        urlstr string
        result string
    }{
        {"http://foo.redhat.com", "foo.redhat.com"},
        {"https://registry.reg-aws.openshift.com:443", "registry.reg-aws.openshift.com:443"},
    }

    for _, testcase := range testcases {
        url, err := url.Parse(testcase)
        if err != nil {
            t.Fatal("Error: ", err)
        }

        ocpa := OpenShiftAdapter{
            Config: Configuration{
                URL: url,
            },
        }
        ft.Equal(t, ocpa.RegistryName(), result, "the registry name returned did not match expected value")
    }
}
```

jianzhangbjz commented:
@dymurray @jmrodri Many thanks for your great feedback. I updated them according to your comments and added the case conversion, please have a review.

jianzhangbjz commented:
@dymurray @jmrodri I removed the case conversion as said in #86.
